src/rez/cli/pip.py

```diff
@@ -29,17 +38,129 @@ def setup_parser(parser, completions=False):
 
 def command(opts, parser, extra_arg_groups=None):
     from rez.pip import pip_install_package, run_pip_command
+    from rez.vendor.version.version import Version, VersionRange
     import sys
+    import json
+    import urllib2
+    import itertools
 
-    if not (opts.search or opts.install):
-        parser.error("Expected one of: --install, --search")
+    if not (opts.search or opts.install or opts.list or opts.query or opts.constrain):
+        parser.error("Expected one of: --install, --search, --list, --query, --constrain")
 
     if opts.search:
         p = run_pip_command(["search", opts.PACKAGE])
         p.wait()
         return
 
-    installed_variants, skipped_variants = pip_install_package(
+    if opts.list:
+        opts.PACKAGE += "=="
+        output = run_pip_command(["install", opts.PACKAGE], process_output=True)
+        for package_version in output:
+            print(package_version)
+        return
+
+    if opts.query:
+        URL = "https://pypi.python.org/pypi/{0}/{1}/json"
+        try:
+            data = json.load(urllib2.urlopen(URL.format(opts.PACKAGE, opts.query)))
+
+            classifiers = data["info"]["classifiers"]
+            python_requires = [classifier for classifier in classifiers if "Programming Language :: Python ::" in classifier]
+            python_requires = ''.join(python_requires)
+            python_requires = python_requires.split("Programming Language :: Python :: ")
+            python_requires = filter(None, python_requires)
+
+            if not python_requires:
+                print("Package version is valid but PyPi does not contain a specific Python version. Too old?")
```

Comment:
Agreed, the reason I said too old is because it normally happens with old packages newer ones seem to have that metadata always defined. This was mostly for debugging purposes so will be happy to have more descriptive messages.